lib/lcdbwf/R/functional_enrichment.R

```diff
@@ -44,7 +44,7 @@ run_enricher <- function(res_list, ontology_list, config,
               TERM2GENE=ontology_list[['term2gene']][[ont]],
               TERM2NAME=ontology_list[['term2name']][[ont]],
               config=config,
-              universe=universe
+              universe=universe_list[[name]]
```

Comment:
Should add a check earlier in this function to ensure that a universe entry exists in the list for all names in `res_list`, in cases where the user provides their own. If a name doesn't exist, I guess give an error? Or is there a sensible fallback behavior?

Reply:
The `universe_list[[name]]` returns `NULL` when `universe_list` was set to `NULL` in the previous chunk (when `config$functional_enrichment$limit_background_genes` being `no_filter`).

```r
> universe_list
NULL
> universe_list[['WT_Dox_vs_WT_Vehicle']]
NULL
```

Then, the `enrich_test` function calls `clusterProfiler::enricher` function where the `universe` argument is set to `NULL`, which will use all genes in each database. 

I will add a step raising an error when custom `universe_list` has different names from `res_list` if it's set by user manually.